catalogs/catalog-jdbc-common/src/main/java/org/apache/gravitino/catalog/jdbc/utils/DataSourceUtils.java

```diff
@@ -76,6 +78,41 @@ private static Properties getProperties(JdbcConfig jdbcConfig) {
     return properties;
   }
 
+  private static void validateJdbcConfig(JdbcConfig jdbcConfig) {
+    String driver = jdbcConfig.getJdbcDriver();
+    String url = jdbcConfig.getJdbcUrl();
+    Map<String, String> all = jdbcConfig.getAllConfig();
+    String lowerUrl = url == null ? "" : url.toLowerCase();
```

Comment:
The URL cannot be null, so the check is redundant.

Reply:
You should URL decode the JDBC URL because sometimes, people can sneakily try to evade checks like this by URL escaping 1 or more chars the URL including the query param names. In Zeppelin, we URL decode the string in a loop in case there is double or triple encoding (or worse).

Reply:
Note that URLDecoder.decode (as used in Zeppelin) may also cause issues, as '+' characters become spaces, and % decoding might affect slashes and colons.

Reply:
The Zeppelin code only URL decodes as part of checking for problematic query parameters. If the URL is accepted as valid, the raw URL is passed to the JDBC Driver.
I have seen a few projects use URL decoding as part of this JDBC URL checks but looking at the related issues and PR descriptions that bring in the URL decoding, little evidence is brought in that the URL decoding is necessary.
I suspect that the evidence could be in mailing lists and deliberately not discussed in the PR.